pom.xml

```diff
@@ -120,6 +120,12 @@
       <version>1.9.5</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>cloudbees-folder</artifactId>
+      <version>4.2.3</version>
+      <scope>test</scope>
+    </dependency>
```

Comment:
To include `Folder`s in test. Unfortunately this seems to have exposed a subtle bug in `maven-hpi-plugin`: we would like `hpi:run` to create `work/plugins/cloudbees-folder.jpi`, which it does, and `target/credentials.hpi!/META-INF/MANIFEST.MF` and `target/test-classes/the.hpl` both correctly omit mention of this dependency (because the scope is `test`), yet `work/plugins/credentials.hpl` includes

    Plugin-Dependencies: ui-samples-plugin:1.526,cloudbees-folder:4.2.3

which causes `hpi:run` to produce the error

```
… hudson.PluginManager$1$3$2$1 reactOnCycle
SEVERE: found cycle in plugin dependencies: (root=Plugin:credentials, deactivating all involved) Plugin:credentials -> Plugin:cloudbees-folder -> Plugin:credentials
```

and refuse to load folder-related extensions.

This seems to have been caused by https://github.com/jenkinsci/maven-hpi-plugin/commit/20e49d0 in combination with https://github.com/jenkinsci/maven-hpi-plugin/commit/7ea2da1. It is unfortunate that the ITs do not test `hpi:run` at all.